src/qibolab/instruments/rfsoc/driver.py

```diff
@@ -90,7 +92,20 @@ def _execute_pulse_sequence(
             "sequence": convert(sequence, qubits, self.sampling_rate),
             "qubits": [asdict(convert(qubits[idx])) for idx in qubits],
         }
-        return client.connect(server_commands, self.host, self.port)
+        try:
+            return client.connect(server_commands, self.host, self.port)
+        except RuntimeError as e:
+            if "exception in readout loop" in str(e):
+                log.warning(
+                    "%s %s",
+                    "Exception in readout loop. Attempting again",
+                    "You may want to increase the relaxation time.",
+                )
+                return client.connect(server_commands, self.host, self.port)
```

Comment:
Are you sure you want to attempt reconnection?

Just failing would be a more minimal and predictable solution, if the user wants multiple attempt he could re-execute (as many times as he wishes).

Reply:
Qick sometimes fails when the program is too fast, in a non-deterministic manner.
If it really throws an error it risks to stop a real program (a sweep, an algorithm), while in this way it should be more robust.

I added this in response to Javier and Sergi complains that the RFSoC was not usable because of this problem

Reply:
But the connection is happening only at the very beginning, so there is no chance to block in the middle any internal program.

Maybe, a better solution for external programs as well (those in which the Qibolab execution is called possibly multiple times, as part of a larger program) would be to reuse the connection, and disconnect only at the end. This could be the case of Qibocal.

If they are writing their own code, and they need to connect and disconnect (because of whatever reason) they could reattempt the execution on their side.

Reply:
The block the RFSoC is subject to happens in the middle of the execution, between shots, and it's not fault of the connection but of qick itself. Re-attempting to launch the program usually works

Reply:
Ok, sorry, I got confused by the name: `client.connect()` is not only the connection method, but also the one executing the code.

Was there a reason to keep the connection together with the execution?
(in the other drivers, we have a separate connection phase)

In any case, I guess at this point the problem was with Qibocal itself, but I wonder if we should find another way to allow rerunning from Qibocal, to expose it to the user (and avoid unexpected behaviors).

Reply:
I think the main problem is QICK instability and its non-deterministic behavior... 
How can I add proper exception handling if the code raises errors in **some** cases?

Reply:
If you could tell me more about *which* cases, maybe I will have more chances to give you a sensible answer xD

However, you're already filtering the errors in the present code. Handling or reraising is definitely an option.

This conversation's point is just about the opportunity of automatic resubmission, the handling part is just fine (maybe your question was beyond the current behavior, but just in case).